### DISCUSSION TITLE
"migration:run" command not found outside of build folder after latest update

### DISCUSSION THREAD

**Discussion by backspacerhino** [adonisjs/core#2167] [Help] [answered]
I've followed the latest three update guides:

https://preview.adonisjs.com/blog/august-2020-release
https://preview.adonisjs.com/blog/october-2020-release
https://preview.adonisjs.com/blog/december-2020-release

and when I try to run anything node ace related I get `"x" command not found`

for example:

`node ace migration:run` results in `[ error ]  "migration:run" command not found`

I am running `node ace serve -w` in another terminal as per instructions:

`All this confusion is a result of a stale build folder and you have to make sure that one process is always running to keep the build output up to date.`

When I run it from build folder it works but my biggest issue is that I can't run `node -r @adonisjs/assembler/build/register japaFile.ts` either outside or inside the build folder

because I receive the next:

```[ error ]  "migration:run" command not found
Error: Command failed with exit code 1: /home/mario/.nvm/versions/node/v12.18.4/bin/node -r @adonisjs/assembler/build/register ace migration:run
    at makeError (/home/mario/Documents/Work/project/node_modules/execa/lib/error.js:59:11)
    at handlePromise (/home/mario/Documents/Work/project/node_modules/execa/index.js:114:26)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  shortMessage: 'Command failed with exit code 1: /home/mario/.nvm/versions/node/v12.18.4/bin/node -r @adonisjs/assembler/build/register ace migration:run',
  command: '/home/mario/.nvm/versions/node/v12.18.4/bin/node -r @adonisjs/assembler/build/register ace migration:run',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: undefined,
  stderr: undefined,
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}```

I've reinstalled entire node_modules, rebuild the build project.

What might be the cause of this?

**Comment by thetutlage** [comment, marked as answer]
Assuming that you have the latest versions installed. You need to do following in order for ace commands to be discovered. 

```sh
node ace generate:manifest
```

And you should be running tests from the project root and not the build folder. The `builder` folder holds the compiled Javascript code and running `node -r @adonisjs/assembler/build/register japaFile.ts` is simply not going to work, coz of the following reasons.

- The `node -r @adonisjs/assembler/build/register japaFile.ts` reference the `japaFile.ts`, there is no `japaFile.ts` inside the build folder.
- The `-r @adonisjs/assembler/build/register` is require hook to compile Typescript code in-memory. The build folder has Javascript code and not the typescript code.
